docs/shared/recipes/ci-deployment.md

```diff
@@ -16,12 +79,95 @@ Some executors automatically generate output `package.json` and the lock file ge
 - `@nx/js:tsc`
 - `@nx/next:build`
 
-## Using a custom executor
+{% /tab %}
+{% /tabs %}
+
+## Programmtic usage
+
+If you using a custom setup that does not support the creation of a `package.json` or a lock file, you can still use Nx to generate them via The `createPackageJson` and `createLockFile` functions which are exported from `@nx/js`:
+
+{% tabs %}
+
+{% tab label="Custom script" %}
+
+If you need to use a custom script, to build your application it should look similar to the following:
 
-If you are using a custom executor or an executor that does not support `generatePackgeJson` or `generateLockfile` flags, you can still use Nx to generate `package.json` and the lock file. The `createPackageJson` and `createLockFile` functions are exported from `@nx/devkit`:
+```javascript {% fileName="scripts/custom-build.js" %}
+const { exec } = require('child_process');
+const {
+  createProjectGraphAsync,
+  readCachedProjectGraph,
+  detectPackageManager,
+} = require('@nx/devkit');
+const {
+  createLockFile,
+  createPackageJson,
+  getLockFileName,
+} = require('@nx/js');
+const { writeFileSync } = require('fs');
+
+async function main() {
+  const outputDir = 'dist'; // You can replace this with the output directory you want to use
+  // Detect the package manager you are using (npm, yarn, pnpm)
+  const pm = detectPackageManager();
+  let projectGraph = readCachedProjectGraph();
+  if (!projectGraph) {
+    projectGraph = await createProjectGraphAsync();
+  }
+  // You can replace <NX_TASK_TARGET_PROJECT> with the name of the project if you want.
+  const projectName = process.env.NX_TASK_TARGET_PROJECT;
+  const packageJson = createPackageJson(projectName, projectGraph, {
+    isProduction: true, // Used to strip any non-prod dependencies
+    root: projectGraph.nodes[projectName].data.root,
+  });
+
+  const lockFile = createLockFile(
+    packageJson,
+    projectGraph,
+    detectPackageManager()
+  );
+
+  const lockFileName = getLockFileName(pm);
+
+  writeFileSync(`${outputDir}/package.json`, packageJson);
+  writeFileSync(`${outputDir}/${lockFileName}`, lockFile, {
+    encoding: 'utf8',
+  });
+
+  //... Any additional build steps you want to run
+
+  exec(`${pm.exec} nx build`, (err, stdout, stderr) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    console.log(stdout);
+  });
+}
+
+main();
+```
+
+Then to run the script, update your `package.json` to include the following:
+
+```json {% fileName="package.json" %}
+{
+  "scripts": {
+    "build": "node scripts/custom-build.js"
```

Comment:
Maybe this should be `custom-build` rather than `build` to differentiate between build targets that could be inferred or configured in `project.json`?

Reply:
Or maybe just `copy-package-json` script name?

Reply:
it is `custom-build`